src/FieldWarning/Assets/Ingame/UI/SpawnPointBehaviour.cs

```diff
@@ -11,55 +11,44 @@
  * the License for the specific language governing permissions and limitations under the License.
  */
 
-using UnityEngine;
 using System.Collections.Generic;
 using System.Linq;
-
-public class SpawnPointBehaviour : MonoBehaviour
-{
-    public PFW.Model.Game.Team Team;
-
+using PFW.Model.Game;
+using UnityEngine;
+public class SpawnPointBehaviour : MonoBehaviour {
     public const float MIN_SPAWN_INTERVAL = 2f;
     public const float QUEUE_DELAY = 1f;
-
-    private Vector3 oldPosition;
-    private Queue<GhostPlatoonBehaviour> spawnQueue = new Queue<GhostPlatoonBehaviour>();
-    private float spawnTime = MIN_SPAWN_INTERVAL;
-
-    public void Awake()
-    {
-        Team = GetComponentInParent<PFW.Model.Game.Team>();
+    public Team @Team { get; private set; }
+    
+    private Queue<GhostPlatoonBehaviour> _spawnQueue { get; } = new Queue<GhostPlatoonBehaviour>();
+    private float _spawnTime = MIN_SPAWN_INTERVAL;
+    
+    public void Awake() {
+        this.Team = this.GetComponentInParent<Team>();
     }
 
-    public void Start()
-    {
-        GetComponentInChildren<Renderer>().material.color = Team.Color;
-
-        Team.Session.AddSpawnPoint(this);
+    public void Start() {
+        this.GetComponentInChildren<Renderer>().material.color = this.Team.Color;
+        this.Team.Session.AddSpawnPoint(this);
     }
 
-    public void Update()
-    {
-        if (!spawnQueue.Any())
-            return;
-        
-        spawnTime -= Time.deltaTime;
-        if (spawnTime > 0)
-            return;
+    public void Update() {
+        // If there is no one in the spawn queue then don't continue.
+        if (!this._spawnQueue.Any()) return;
+
+        // Check if spawn time minus the current deltaTime is > 0 then exit method;
+        if (this._spawnTime -= Time.deltaTime > 0) return;
 
-            
-        GhostPlatoonBehaviour ghostPlatoon = spawnQueue.Dequeue();
-        ghostPlatoon.Spawn(transform.position);
+        // NullReferenceException City - position is not set properly.
+        this._spawnQueue.Dequeue().Spawn(this.transform.position);
 
-        if (spawnQueue.Count > 0)
-            spawnTime += MIN_SPAWN_INTERVAL;
-        else
-            spawnTime = QUEUE_DELAY;
-        
+        // Increase spawn queue time if total amount in queue is greater than zero
+        //  otherwise set to QUEUE_DELAY (1f) PS: Turnary > if statements
+        this._spawnTime = (this._spawnQueue.Count > 0) ?
+            this._spawnTime += MIN_SPAWN_INTERVAL : QUEUE_DELAY;
     }
 
-    public void BuyPlatoons(List<GhostPlatoonBehaviour> ghostPlatoons)
-    {
-        ghostPlatoons.ForEach(x => spawnQueue.Enqueue(x));
+    public void BuyPlatoons(List<GhostPlatoonBehaviour> ghostPlatoons) {
+        ghostPlatoons.ForEach (x => this._spawnQueue.Enqueue(x));
```

Comment:
Against the style guide. If you are proposing an intentional change then lets talk on discord.

Reply:
I personally didn't touch that. I'm sure ReSharper did. But yes, I'd like to chat with you about some changes to the style guide.